app/src/lib/stores/app-store.ts

```diff
@@ -1078,14 +1079,26 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
     setNumber(LastSelectedRepositoryIDKey, repository.id)
 
-    if (repository.missing) {
+    // if repository might be marked missing, try checking if it has been restored
+    const refreshedRepository = await this.recoverMissingRepository(repository)
+    if (refreshedRepository.missing) {
       // as the repository is no longer found on disk, cleaning this up
       // ensures we don't accidentally run any Git operations against the
       // wrong location if the user then relocates the `.git` folder elsewhere
       this.gitStoreCache.remove(repository)
       return Promise.resolve(null)
     }
 
+    return this._selectRepository_refreshTasks(
+      refreshedRepository,
+      previouslySelectedRepository
+    )
+  }
+
+  private async _selectRepository_refreshTasks(
```

Comment:
we tend to use camelCase for class methods so could `_selectRepository_refreshTasks` be `_selectRepositoryRefreshTasks`? I like that we are encapsulating this logic into its own method